spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/annotation/SleuthSpanCreatorAspectTests.java

```diff
@@ -0,0 +1,345 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.annotation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.assertj.core.api.BDDAssertions;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanReporter;
+import org.springframework.cloud.sleuth.Tracer;
+import org.springframework.cloud.sleuth.annotation.SleuthSpanCreatorAspectTests.TestConfiguration;
+import org.springframework.cloud.sleuth.assertions.ListOfSpans;
+import org.springframework.cloud.sleuth.sampler.AlwaysSampler;
+import org.springframework.cloud.sleuth.util.ArrayListSpanAccumulator;
+import org.springframework.cloud.sleuth.util.ExceptionUtils;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import static org.springframework.cloud.sleuth.assertions.SleuthAssertions.then;
+
+@SpringBootTest(classes = TestConfiguration.class)
+@RunWith(SpringJUnit4ClassRunner.class)
+public class SleuthSpanCreatorAspectTests {
+	
+	@Autowired TestBeanInterface testBean;
+	@Autowired Tracer tracer;
+	@Autowired ArrayListSpanAccumulator accumulator;
+	
+	@Before
+	public void setup() {
+		ExceptionUtils.setFail(true);
+		this.accumulator.clear();
+	}
+	
+	@Test
+	public void shouldCreateSpanWhenAnnotationOnInterfaceMethod() {
+		this.testBean.testMethod();
+		
+		List<Span> spans = new ArrayList<>(this.accumulator.getSpans());
+		then(new ListOfSpans(spans)).hasSize(1).hasASpanWithName("test-method");
+		then(ExceptionUtils.getLastException()).isNull();
+	}
+	
+	@Test
+	public void shouldCreateSpanWhenAnnotationOnClassMethod() {
+		this.testBean.testMethod2();
+
+		List<Span> spans = new ArrayList<>(this.accumulator.getSpans());
+		then(new ListOfSpans(spans)).hasSize(1).hasASpanWithName("test-method2");
+		then(ExceptionUtils.getLastException()).isNull();
+	}
+	
+	@Test
+	public void shouldCreateSpanWithCustomNameWhenAnnotationOnClassMethod() {
+		this.testBean.testMethod3();
+
+		List<Span> spans = new ArrayList<>(this.accumulator.getSpans());
+		then(new ListOfSpans(spans)).hasSize(1).hasASpanWithName("custom-name-on-test-method3");
+		then(ExceptionUtils.getLastException()).isNull();
+	}
+	
+	@Test
+	public void shouldCreateSpanWithCustomNameWhenAnnotationOnInterfaceMethod() {
+		this.testBean.testMethod4();
```

Comment:
is it possible to declare method local classes? that might make what the tests are proving easier to see without navigation.